### PR TITLE
 [Heartbeat] Support map syntax for throttling browser monitors

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes https://github.com/elastic/beats/issues/31362
  
Note, the config for throttling is a bit weird and doesn't really fit with our regular way of configuration, which would be to avoid
`interface{}` and deserialize to a struct. However, I spent a good bit of time struggling to get ucfg to work with our current format + this one, and whipped this implementation out rather quickly, so I'm OK leaving this as-is for a while.

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Run this sample config, and look for log lines in the output like: `Running command: /home/andrewvc/.local/share/nvm/v16.13.2/bin/elastic-synthetics elastic-synthetics --screenshots on --throttling {"download":1.6,"latency":150,"upload":0.75} --inline --rich-events in directory: ''`

```yml
- type: browser
  enabled: true
  id: browser-inline
  name: browser-inline
  throttling:
    download: 1.6
    upload: 0.75
    latency: 150
  source:
    inline:
      script:
        step("load homepage", async () => {
            await page.goto('https://www.elastic.co');
        });
        step("hover over products menu", async () => {
            await page.hover('css=[data-nav-item=products]');
        });
        step("failme", async () => {
            await page.hhover('css=[data-nav-item=products]');
        });
  schedule: "@every 1m"
```

### CODE REVIEW

**Comment by andrewvc** [comment]
@mergifyio update

File: heartbeat/docs/monitors/monitor-browser.asciidoc [hunk 1]
```diff
@@ -189,7 +189,19 @@ run with 5Mbps download, 3Mbps upload and 20ms latency which emulates a standard
...
 Users can control the throttling parameters, Below is an example of emulating a
-3G connection with 1.6Mbps download, 750Kbps upload and 150ms round trip time.
+3G connection with 1.6Mbps download, 750Kbps upload and 150ms round trip time
+in two different supported formats.
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: browser
+  schedule: '@every 1m'
+  throttling: "1.6d/0.75u/150l"
```
Comment (vigneshshanmugam) [inline heartbeat/docs/monitors/monitor-browser.asciidoc:199]: IMO, we should remove the old format and references so we can do breaking change on the Synthetics agent from next version. Would give us enough room to move to GA. WDYT?

**Review by vigneshshanmugam — APPROVED** [review]
LGTM expect for the deprecation part. 

Verified with the latest Synthetics agent. 

```
"payload":{"network_conditions":{"uploadThroughput":6.5536e+06,"latency":15,"offline":false,"downloadThroughput":1.31072e+07}}
```

File: heartbeat/docs/monitors/monitor-browser.asciidoc (hunk 1 above)
Comment (andrewvc) [reply heartbeat/docs/monitors/monitor-browser.asciidoc:199]: I'm not nuts about removing it, though I'd be fine with deprecating it, which we can do in a follow-up.

I realize we're in beta so we could just remove it, but for people who are using it that's rude, and it's esp. tricky for configs serialized in fleet. I think we need to have discuss what changing config options looks like in our new world of GUI configs.

Comment (paulb-elastic) [reply heartbeat/docs/monitors/monitor-browser.asciidoc:199]: @andrewvc I know this has already been merged, but the _new format_ seems to still show the old one too (in `throttling: "1.6d/0.75u/150l"`), shouldn't the new one be:

```
throttling:
  download: 1.6
  upload: 0.75
  latency: 150
```

Comment (vigneshshanmugam) [reply heartbeat/docs/monitors/monitor-browser.asciidoc:199]: Currently, HB supports both formats and we have not deprecated the old one which could be the reason.

Comment (andrewvc) [reply heartbeat/docs/monitors/monitor-browser.asciidoc:199]: I've created a PR to fix this, good catch @paulb-elastic https://github.com/elastic/beats/pull/31738/files

**Pull request merged**